worker/dbaccessor/worker_notcgo.go

```diff
@@ -0,0 +1,55 @@
+// +build !cgo
+
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package dbaccessor
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/juju/errors"
+)
+
+type noopApp struct{}
```

Comment:
Should this perhaps be an unsupported or incompatible app instead? No-op is a bit ambiguous given that `Open` always errors.